lib/Migration/Version4000Date20241108004215.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Nextcloud - ContextChat
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author Anupam Kumar <kyteinsky@gmail.com>
+ * @copyright Anupam Kumar 2024
+ */
+
+declare(strict_types=1);
+
+namespace OCA\ContextChat\Migration;
+
+use Closure;
+use OCA\ContextChat\BackgroundJobs\IndexerJob;
+use OCA\ContextChat\BackgroundJobs\InitialContentImportJob;
+use OCA\ContextChat\BackgroundJobs\SchedulerJob;
+use OCA\ContextChat\BackgroundJobs\StorageCrawlJob;
+use OCA\ContextChat\BackgroundJobs\SubmitContentJob;
+use OCA\ContextChat\Service\ProviderConfigService;
+use OCP\BackgroundJob\IJobList;
+use OCP\DB\Exception;
+use OCP\IDBConnection;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+
+class Version4000Date20241108004215 extends SimpleMigrationStep {
+
+	public function __construct(
+		private IDBConnection $db,
+		private IJobList $jobList,
+		private ProviderConfigService $providerConfigService,
+	) {
+	}
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 *
+	 * @return void
+	 */
+	public function postSchemaChange(IOutput $output, Closure $schemaClosure, array $options) {
+		$output->startProgress(10);
+		$this->jobList->remove(SchedulerJob::class);
+		$output->advance(1);
+		$this->jobList->remove(StorageCrawlJob::class);
+		$output->advance(1);
+		$this->jobList->remove(IndexerJob::class);
+		$output->advance(1);
+		$this->jobList->remove(InitialContentImportJob::class);
+		$output->advance(1);
+		$this->jobList->remove(SubmitContentJob::class);
+		$output->advance(1);
+
+		try {
+			$qb = $this->db->getQueryBuilder();
+			$qb->delete('context_chat_delete_queue')->executeStatement();
+		} catch (Exception $e) {
+			$output->warning($e->getMessage());
+		}
+		$output->advance(1);
+
+		try {
+			$qb = $this->db->getQueryBuilder();
+			$qb->delete('context_chat_content_queue')->executeStatement();
+		} catch (Exception $e) {
+			$output->warning($e->getMessage());
+		}
+		$output->advance(1);
+
+		try {
+			$qb = $this->db->getQueryBuilder();
+			$qb->delete('context_chat_queue')->executeStatement();
+		} catch (Exception $e) {
+			$output->warning($e->getMessage());
+		}
+		$output->advance(1);
+
+		$appIds = [];
+		foreach ($this->providerConfigService->getProviders() as $providerKey => $props) {
+			[$appId, $providerId] = explode('__', $providerKey, 2);
+			$this->providerConfigService->removeProvider($appId, $providerId);
+			$appIds[$appId] = true;
+		}
```

Comment:
just setting `providers` app config to an empty string should be enough. Application.php should handle the rest.

Reply:
ah, makes sense